.github/workflows/run-tests.yaml

```diff
@@ -0,0 +1,27 @@
+name: Run Unit Tests
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Set up Golang
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.21'
```

Comment:
i dont know if this affects anything, but the go version we have in the go.mod is 1.22.3 does go-version in ci have to match that or now?